docs/gadget-devel/metrics.md

```diff
@@ -74,6 +74,23 @@ values.
 
 > Note: adding multiple metric.key annotations will increase cardinality a lot and take up more space and storage.
 
+#### Adding an implicit counter
```

Comment:
We need to document this also for gadget users on top of https://github.com/inspektor-gadget/inspektor-gadget/pull/3539, but it's fine if you want to do it on a follow-up PR, once both are merged.